### PR TITLE
#002 Ft user registration 

### PR DESCRIPTION
**Pull request by buka1calvin**
**What does this PR do?**

- Implement user registration 

**Description**

-  As any user

- A user should be able to register

**How should this be tested manually?**

- Clone this repo and checkout to ft-user-registration-002 branch
- Install all dependencies by running "npm install"
- run the server with "npm run dev" command
- Then register 

** Any background you want to provide?**
- None

**Screenshot**
- None

**Questions**
- None

### CODE REVIEW

**Review by nkurunziza1 — COMMENTED** [review]
we need to add the phone country code

**Review by nkurunziza1 — COMMENTED** [review]
we need to add the phone country code

**Comment by buka1calvin** [comment]
> we need to add the phone country code

ok done

**Review by dean-daryl — CHANGES_REQUESTED** [review]
Please add validations ❗❗

**Review by dean-daryl — APPROVED** [review]

**Review by nkurunziza1 — APPROVED** [review]

File: src/resolvers/loginUserResolver.ts [hunk 1]
```diff
@@ -1,134 +1,119 @@
...
-        path: 'role',
-        populate: {
-          path: 'permissions',
-          model: PermissionModel, 
-        },
-      });
-      return users;
+      // if (!ctx.currentUser) {
+      //   throw new AuthenticationError("You must be logged in");
+      // }
```
Comment (Anguandia) [inline src/resolvers/loginUserResolver.ts]: Good work @buka1calvin, please remove commented-out code, and check throughout that you do not have any recurrence of commented code

**Review by Anguandia — CHANGES_REQUESTED** [review]
Please Squash your commits to 2 at most

File: src/resolvers/loginUserResolver.ts (hunk 1 above)
Comment (buka1calvin) [reply src/resolvers/loginUserResolver.ts]: ok sir let ,me work on it

**Review by Anguandia — APPROVED** [review]

**Review by JENNY-257 — APPROVED** [review]

**Pull request merged**